commands/connect/db-set.js

```diff
@@ -32,13 +32,13 @@ module.exports = {
   needsApp: true,
   needsAuth: true,
   run: cli.command(co.wrap(function * (context, heroku) {
-    context.region = yield regions.determineRegion(context, heroku)
     let data = {
       db_key: context.flags.db,
       schema_name: context.flags.schema
     }
 
     let connection = yield api.withConnection(context, heroku)
+    context.region = connection.region_url
```

Comment:
This is the most ubiquitous change in this PR. Since the `connection` now conveys its own `region_url`, it makes sense to set `context.region` to this value.